PEM-7798: auto cluster role bindings (#7750)
spectrocloud/librarium: docs/docs-content/clusters/cluster-management/platform-settings/cluster-auto-rbac.md

---
sidebar_label: "Automatic Cluster Role Bindings"
title: "Automatic Cluster Role Bindings"
description: "Learn about Palette's Automatic Cluster Role Bindings platform setting."
hide_table_of_contents: false
sidebar_position: 20
tags: ["clusters", "cluster management"]
---

> Palette uses Automatic Cluster Role Bindings as a way to automatically apply appropriate cluster role bindings to clusters. When Automatic Cluster Role Bindings is enabled, any clusters created with Palette Identity Provider (IDP) enabled will automatically receive Role-Based Access Control (RBAC) bindings. This feature is disabled by default and activated at the tenant level. When configured, cluster role bindings are automatically applied to all clusters created and deployed at the tenant and project levels. 
> 
> Any clusters that exist prior to enabling **Automatic Cluster Role Bindings** will receive the applicable role bindings as part of a system scheduler job that runs every 15 minutes. The scheduler also will update RBACs if user permissions have changed, a new user is added, or a user is removed.
> 
> Automatic Cluster Role Bindings correspond to the standard Kubernetes cluster roles `cluster-admin`, `cluster-edit`, and `cluster-view` and are assigned based on the user's roles in Palette. For more information about user roles, refer to [Roles and Permissions](../../../user-management/palette-rbac/palette-rbac.md). For more information about using Palette as an IDP, refer to [SAML and OIDC SSO](../../../user-management/saml-sso/saml-sso.md).

Reviewer: ```suggestion
**Automatic Cluster Role Bindings** is a feature in Palette that automatically applies the appropriate Kubernetes cluster role bindings to clusters based on user roles. This ensures that Role-Based Access Control (RBAC) permissions are consistently applied without requiring manual configuration.

When **Automatic Cluster Role Bindings** is enabled, any clusters created with Palette Identity Provider (IDP) integration will automatically receive the correct RBAC bindings. These are applied based on the user's role in Palette and correspond to the standard Kubernetes cluster roles `cluster-admin`, `cluster-edit`, and `cluster-view`.

**Automatic Cluster Role Bindings** is disabled by default and can be enabled at the tenant level. Once enabled, cluster role bindings are automatically applied to all newly created clusters, whether they are provisioned at the tenant or project level.

Clusters created before enabling **Automatic Cluster Role Bindings** are updated via a system scheduler job that runs every 15 minutes. The scheduler also will update RBACs if user permissions are changed, a new user is added, or a user is removed.

For more information about user roles, refer to [Roles and Permissions](../../../user-management/palette-rbac/palette-rbac.md). For more information about using Palette as an IDP, refer to [SAML and OIDC SSO](../../../user-management/saml-sso/saml-sso.md).
```

## Prerequisites

- Tenant admin access to Palette.

- An existing cluster profile with Palette eXtended Kubernetes with Palette set as the OIDC Identity Provider. For steps on creating a cluster profile, refer to our [Create Cluster Profiles](../../../profiles/cluster-profiles/create-cluster-profiles/create-cluster-profiles.md) guide.

## Enablement

1. Log in to [Palette](https://console.spectrocloud.com) as a tenant administrator.

2. From the left main menu, select **Tenant Settings**.

3. From the **Tenant Settings Menu**, below **Platform**, select **Platform Settings**.

> 4. Toggle the **Automatic Cluster Role Bindings** button.

Reviewer: ```suggestion
4. Toggle the **Automatic Cluster Role Bindings** button to activate the feature.
```

---
sidebar_label: "Automatic Cluster Role Bindings"
title: "Automatic Cluster Role Bindings"
description: "Learn about Palette's Automatic Cluster Role Bindings platform setting."
hide_table_of_contents: false
sidebar_position: 20
tags: ["clusters", "cluster management"]
---

> Palette provides Automatic Cluster Role Bindings as a way to automatically add appropriate cluster role bindings. When Automatic Cluster Role Bindings is enabled, newly created clusters that have Palette IDP enabled will have RBAC bindings set. This feature is set at the Tenant level but enables role binding at both Tenant and Project levels. The binding applied will be equivalent to the Kubernetes cluster role of `cluster-admin`, `cluster-edit`, or `cluster-view`. This will be based on the Palette user roles assigned. For more information about user roles, refer to [Roles and Permissions](../../../user-management/palette-rbac/palette-rbac.md). 

Reviewer: ```suggestion
Palette uses Automatic Cluster Role Bindings as a way to automatically apply appropriate cluster role bindings to new clusters. When Automatic Cluster Role Bindings is enabled, any newly created clusters with Palette Identity Provider (IDP) enabled will automatically receive Role-Based Access Control (RBAC) bindings. This feature is disabled by default and activated at the tenant level. When configured, cluster role bindings are automatically applied to all clusters created at the tenant and project levels.

Automatic Cluster Role Bindings correspond to the standard Kubernetes cluster roles `cluster-admin`, `cluster-edit`, and `cluster-view` and are assigned based on the user's roles in Palette. For more information about user roles, refer to [Roles and Permissions](../../../user-management/palette-rbac/palette-rbac.md). 
```

The automatic role binding cannot be edited as this is created by the Palette system.

> Turning off this feature will disable automatic role binding. 

Reviewer: I tried to incorporate "Turning off this feature will disable automatic role binding" in the intro by stating it was disabled by default. (Correct me if I'm wrong--that's what I gathered from the jira.)

Do we need "The automatic role binding cannot be edited as this is created by the Palette system"? It's just a toggle, so it seems self-explanatory you cant modify the default config.

:::info

> If clusters exist before enabling Automatic Cluster Role Binding, they will get the role bindings as part of a system scheduler job that runs every 15 minutes. The scheduler also will update RBACs if user permissions have changed, a new user is added, or a user is removed.

Reviewer: ```suggestion
Any clusters that exist prior to enabling **Automatic Cluster Role Bindings** will receive the applicable role bindings as part of a system scheduler job that runs every 15 minutes. The scheduler also will update RBACs if user permissions have changed, a new user is added, or a user is removed.
```

Reviewer: Seems a little odd to put this in an admonition when it's default behavior for the feature? Maybe find a way to incorporate it into the intro?

:::

## Prerequisites

- Tenant admin access to Palette.

> - An existing profile with Palette eXtended Kubernetes with Palette set as the OIDC Identity Provider. For steps on creating a cluster profile, refer to [Cluster Profile](../../../profiles/cluster-profiles/cluster-profiles.md) to get started.

Reviewer: ```suggestion
- An existing cluster profile with Palette eXtended Kubernetes with Palette set as the OIDC Identity Provider. For steps on creating a cluster profile, refer to our [Create Cluster Profiles](../../../profiles/cluster-profiles/create-cluster-profiles/create-cluster-profiles.md) guide.
```
This one might work better. Puts them a little closer to the "create" process :)

Reviewer: Maybe we can also link them to https://docs.spectrocloud.com/integrations/packs/?pack=kubernetes for how to enable OIDC? Remember to use a versioned link, though!

## Enablement

> 1. Log in to [Palette](https://console.spectrocloud.com) as a Tenant administrator.

Reviewer: ```suggestion
1. Log in to [Palette](https://console.spectrocloud.com) as a tenant administrator.
```

> 2. From the left **Main Menu**, select **Platform Settings**.
> 
> 3. Toggle the **Automatic Cluster Role Binding** button.
> 
> 4. A pop-up box will ask you to confirm the action. Click **OK**.

Reviewer: ```suggestion
2. From the left main menu, select **Tenant Settings**.

3. From the **Tenant Settings Menu**, below **Platform**, select **Platform Settings**. 

4. Toggle the **Automatic Cluster Role Bindings** button.

5. A pop-up box prompts you to confirm the action. Click **OK**.
```

Reviewer: Formatting updates + addition of "Tenant Settings" selection. (We always lowercase "left main menu" and do not bold it--the formatting you used is from our old style :))

## Validate

> 1. Log in to [Palette](https://console.spectrocloud.com).
> 
> 2. Navigate to the left **Main Menu** and click on **Project Settings**.
> 
> 3. Select **Platform Settings**.
> 
> 4. The **Cluster Auto Remediation** toggle button is checked.

Reviewer: ```suggestion
1. Log in to [Palette](https://console.spectrocloud.com) as a tenant administrator.

2. From the left main menu, select **Tenant Settings**.

3. From the **Tenant Settings Menu**, below **Platform**, select **Platform Settings**. 

4. Ensure the **Automatic Cluster Role Bindings** toggle is set to active.
```

5. Deploy a cluster with Palette IDP enabled and no RBAC cluster bindings set.

> 6. Navigate to the left **Main Menu**, select **Audit Logs**.

Reviewer: ```suggestion
6. From the left main menu, select **Audit Logs**.
```

> 7. Set the filter **Log Type** to **Update** and **Resource Type** to **Cluster**. 

Reviewer: ```suggestion
7. Set the filter **Log Type** to **Update** and **Resource Type** to **Cluster**. Look for entries that indicate RBAC has been updated.
```
